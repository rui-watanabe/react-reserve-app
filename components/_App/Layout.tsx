import React from 'react';
import Head from 'next/head';
import { Container } from 'semantic-ui-react';

import { Header } from './Header';
import { HeadContent } from './HeadContent';
import { HeaderProps } from './HeaderType';

export function Layout({
  children,
  user,
}: React.PropsWithChildren<HeaderProps>) {
  return (
    <>
      <Head>
        <HeadContent />
        {/* Stylesheets */}
        <link rel="stylesheet" type="text/css" href="/static/styles.css" />
        <link rel="stylesheet" type="text/css" href="/static/nprogress.css" />
        <link
          rel="stylesheet"
          href="//cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.2.2/semantic.min.css"
        />
        <title>MernWithTsApp</title>
      </Head>
      <Header user={user} />
      <Container text style={{ paddingTop: '1em' }}>
        {children}
      </Container>
    </>
  );
}
