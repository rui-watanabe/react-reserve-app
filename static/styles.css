:root {
  --bodyBg: #f0f4f8;
  --menuBg: #008080;
  --progressBar: #54d1db;
}

body {
  background: var(--bodyBg);
}

#menu {
  background: var(--menuBg);
}

/* Overrides default styled accordion styles (too much bottom padding) */
.ui.styled.accordion .content {
  margin: 0;
  padding: 0.5em 1em 0.75em !important;
}

.ui.card > .image:not(.ui) > img, .ui.cards > .card > .image:not(.ui) > img  {
  height: 212px;
}
