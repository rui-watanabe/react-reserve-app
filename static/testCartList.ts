import { CartProductType } from '../src/pagesTypes/CartType';
import { testProductList } from './testProductList';

export const testCartList: CartProductType = [
  {
    quantity: 1,
    product: testProductList[0],
  },
  {
    quantity: 2,
    product: testProductList[1],
  },
];
