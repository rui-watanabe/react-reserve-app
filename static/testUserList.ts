import { ObjectId } from 'mongodb';
import { UserModelType } from '../models/Types/UserType';

export const testUserList: UserModelType[] = [
  {
    _id: new ObjectId('000000000000000000000001'),
    createdAt: new Date('2020-9-22'),
    updatedAt: new Date('2020-9-22'),
    name: 'test',
    email: 'test@gmail.com',
    password: '11111111',
    role: 'root',
  },
  {
    _id: new ObjectId('000000000000000000000002'),
    createdAt: new Date('2020-9-22'),
    updatedAt: new Date('2020-9-22'),
    name: 'test2',
    email: 'test2@gmail.com',
    password: '22222222',
    role: 'admin',
  },
  {
    _id: new ObjectId('000000000000000000000003'),
    createdAt: new Date('2020-9-22'),
    updatedAt: new Date('2020-9-22'),
    name: 'test',
    email: 'test@gmail.com',
    password: '33333333',
    role: 'user',
  },
];
