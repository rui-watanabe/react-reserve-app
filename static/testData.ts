import { ProductModelType } from '../models/Types/ProductType';
import { UserModelType } from '../models/Types/UserType';
import { CartProductType } from '../src/pagesTypes/CartType';
import { testProductList } from './testProductList';
import { testUserList } from './testUserList';

export const testProduct: ProductModelType = testProductList[0];

export const testTotalPages = Math.floor(testProductList.length / 9);

export const testUser: UserModelType = testUserList[2]; // 'user' role user

export const testCartProductList: CartProductType = [
  { product: { ...testProduct }, quantity: 1 },
  { product: { ...testProductList[1] }, quantity: 2 },
  { product: { ...testProductList[2] }, quantity: 3 },
];
