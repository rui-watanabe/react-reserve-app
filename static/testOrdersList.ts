import { ObjectId, ObjectID } from 'mongodb';
import { OrderModelType } from '../models/Types/OrderType';
import { testProductList } from './testProductList';
import { testUserList } from './testUserList';

export const testOrdersList: OrderModelType[] = [
  {
    _id: new ObjectId('200000000000000000000001'),
    createdAt: new Date(2020, 4, 1, 12, 30, 45),
    updatedAt: new Date(2020, 4, 1, 12, 30, 45),
    products: [
      {
        _id: new ObjectID('100000000000000000000001'),
        quantity: 1,
        product: testProductList[0],
      },
      {
        _id: new ObjectID('100000000000000000000002'),
        quantity: 2,
        product: testProductList[1],
      },
    ],
    user: testUserList[2]._id, // 'user' role user
    email: testUserList[2].email,
    total: testProductList[0].price + testProductList[1].price * 2,
  },
  {
    _id: new ObjectID('200000000000000000000002'),
    createdAt: new Date(2020, 4, 1, 12, 30, 45),
    updatedAt: new Date(2020, 4, 1, 12, 30, 45),
    products: [
      {
        _id: new ObjectID('100000000000000000000003'),
        quantity: 1,
        product: testProductList[2],
      },
      {
        _id: new ObjectID('100000000000000000000004'),
        quantity: 2,
        product: testProductList[3],
      },
    ],
    user: testUserList[2]._id, // 'user' role user
    email: testUserList[2].email,
    total: testProductList[2].price + testProductList[3].price * 2,
  },
  {
    _id: new ObjectID('200000000000000000000003'),
    createdAt: new Date(2020, 4, 1, 12, 30, 45),
    updatedAt: new Date(2020, 4, 1, 12, 30, 45),
    products: [
      {
        _id: new ObjectID('100000000000000000000005'),
        quantity: 1,
        product: testProductList[4],
      },
      {
        _id: new ObjectID('100000000000000000000006'),
        quantity: 2,
        product: testProductList[5],
      },
    ],
    user: testUserList[2]._id, // 'user' role user
    email: testUserList[2].email,
    total: testProductList[4].price + testProductList[5].price * 2,
  },
];
