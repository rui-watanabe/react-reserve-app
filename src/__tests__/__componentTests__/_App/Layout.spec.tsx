import React from 'react';
import renderer from 'react-test-renderer';
import Layout from '../../../components/_App/Layout';
import { testUser } from '../../../../static/testData';

jest.mock('next/router', () => ({
  events: {
    on: jest.fn(),
    off: jest.fn(),
    emit: jest.fn(),
  },
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      events: {
        on: jest.fn(),
        off: jest.fn(),
        emit: jest.fn(),
      },
    };
  },
}));
test('Layout', () => {
  const component = renderer.create(<Layout user={testUser} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
