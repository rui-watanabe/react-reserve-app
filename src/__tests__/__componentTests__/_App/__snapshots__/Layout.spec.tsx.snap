// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout 1`] = `
Array [
  <div
    className="ui olive fluid inverted stackable menu"
    id="menu"
  >
    <div
      className="ui text container"
    >
      <a
        className="header item"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <img
          className="ui mini image"
          src="/static/logo.svg"
          style={
            Object {
              "marginRight": "1em",
            }
          }
        />
        MERN Reserve
      </a>
      <a
        className="header item"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <i
          aria-hidden="true"
          className="cart large icon"
          onClick={[Function]}
        />
        Cart
      </a>
      <a
        className="header item"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <i
          aria-hidden="true"
          className="user large icon"
          onClick={[Function]}
        />
        Account
      </a>
      <a
        className="header item"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="sign out large icon"
          onClick={[Function]}
        />
        Logout
      </a>
    </div>
  </div>,
  <div
    className="ui text container"
    style={
      Object {
        "paddingTop": "1em",
      }
    }
  />,
]
`;
